pkg/scheduler/placement/provided_rule.go

```diff
@@ -55,9 +56,10 @@ func (pr *providedRule) initialise(conf configs.PlacementRule) error {
 func (pr *providedRule) placeApplication(app *objects.Application, queueFn func(string) *objects.Queue) (string, error) {
 	// since this is the provided rule we must have a queue in the info already
 	queueName := app.GetQueuePath()
-	if queueName == "" {
+	if queueName == "" || common.IsRecoveryQueue(queueName) {
```

Comment:
The queue name can have a prefix of `root.` or might not have it. I can even use provided with a parent rule to and make it really tricky to detect that it would end up in the recovery queue.
A simple ACL check on the recovery queue can prevent all that from being possible.

Reply:
This does seem cleaner. However, how do we enforce a static ACL on the recovery queue? The ACLs get populated via configuration, and since that config won't exist for the recovery queue, the ACLs will be empty.

Reply:
Fixed this via using an impossible ACL for the recovery queue ; all rules (except) recovery now return a flag indicating that queue ACLs need to be checked. The recovery rule returns false, allowing the normal ACL checks to be skipped and the recovery queue used.